### PR TITLE
Add to_date Spark function

### PR DESCRIPTION
**Pull request by CLTFOREVER**

### CODE REVIEW

File: velox/docs/functions/spark/datetime.rst [hunk 1]
```diff
@@ -137,3 +137,8 @@ These functions support TIMESTAMP and DATE input types.
 .. spark:function:: year(x) -> integer
 
     Returns the year from ``x``.
+
+.. spark:function:: to_date(date) -> date
```
[the remark this answers is not in the document]
Comment (CLTFOREVER) [reply velox/docs/functions/spark/datetime.rst:141]: yes，anything conflict?

**Comment by CLTFOREVER** [comment]
@mbasmanova
can you help to review this PR? Thanks.

File: velox/docs/functions/spark/datetime.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (xy2953396112) [reply velox/docs/functions/spark/datetime.rst:141]: ` to_date` function should support two parameters, the second arg_type is format.

**Review by mbasmanova — COMMENTED** [review]
@rui-mo Rui, would you help review this PR?

File: velox/docs/functions/spark/datetime.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (CLTFOREVER) [reply velox/docs/functions/spark/datetime.rst:141]: yes,we have modify  to_date function  internally,I will re implement it according to the community's approach.Thanks

**Review by rui-mo — COMMENTED** [review]
Looks like Spark creates a replacement for this function by using a combination of
`GetTimestamp` and `Cast (from timestamp to date)`. That's means what we got from Spark plan are these two functions, instead of `to_date`.
Given this point, maybe we can double-check the support of this function. BTW, @PHILO-HE has a made a draft for the implementation of `GetTimestamp`. Thanks.

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2044-L2055

**Comment by CLTFOREVER** [comment]
@rui-mo Thanks for your reply, the latest implementation of to_date function in spark is indeed by using a combination of
GetTimestamp and Cast (from timestamp to date). As far as I can see, the function castToDate is ready. If the function GetTimestamp is also ready, the to_date function appears to be completed. So does GetTimestamp have an expected readiness time

**Comment by rui-mo** [comment]
> So does GetTimestamp have an expected readiness time

@CLTFOREVER Thanks for checking. cc @PHILO-HE

File: velox/docs/functions/spark/datetime.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (xy2953396112) [reply velox/docs/functions/spark/datetime.rst:141]: Is there any code update for the `time format`?

**Comment by PHILO-HE** [comment]
Hi @CLTFOREVER, FYI, we created a PR for supporting get_timestamp: https://github.com/facebookincubator/velox/pull/8004.

**Pull request closed without merging**